python/cuml/manifold/umap.pyx

```diff
@@ -417,29 +421,37 @@ class UMAP(Base):
                                                      if convert_dtype
                                                      else None))
             fit(handle_[0],
-                < float*> x_raw,
-                < float*> y_raw,
-                < int > self.X_m.shape[0],
-                < int > self.X_m.shape[1],
-                < UMAPParams*>umap_params,
-                < float*>embed_raw)
+                <float*> x_raw,
+                <float*> y_raw,
+                <int> self.n_rows,
+                <int> self.n_dims,
+                <UMAPParams*>umap_params,
+                <float*>embed_raw)
 
         else:
 
             fit(handle_[0],
-                < float*> x_raw,
-                < int > self.X_m.shape[0],
-                < int > self.X_m.shape[1],
-                < UMAPParams*>umap_params,
-                < float*>embed_raw)
+                <float*> x_raw,
+                <int> self.n_rows,
+                <int> self.n_dims,
+                <UMAPParams*>umap_params,
+                <float*>embed_raw)
 
         self.handle.sync()
 
         return self
 
-    def fit_transform(self, X, y=None, convert_dtype=False):
+    def fit_transform(self, X, y=None, convert_dtype=True):
         """Fit X into an embedded space and return that transformed
         output.
+
+        There is a subtle difference between calling fit_transform(X)
+        and calling fit().transform(). Calling fit_transform(X) will
+        train the embeddings on X and return the embeddings. Calling
+        fit(X).transform(X) will train the embeddings on X and then
+        run a second optimization
+        return the embedding after it is trained while calling
```

Comment:
This is a useful note... do we want this behavior though? Is there a use case for it? It seems surprising to me and would presumably be easy to fix by just running fit

Reply:
I held off on including this behavior in the cuml UMAP implementation until now. Unfortunately, since the reference implementation does this, it has confused users quite a bit that the cuML version behaves differently. In fact, most users don’t realize this is happening under the hood and all they see is that the reference implementation provides exact embeddings in this case and cuml does not. 

I used copy to host because I didn’t see a good way to hash GPU arrays and it seemed like a reasonable near-term solution. 

#963

Reply:
@JohnZed I made this optional and added some documentation on how to enable it, as well as why it exists and why it is not enabled by default. Please let me know if this works for you.